pkg/lib/cockpit-components-dialog.jsx

```diff
@@ -209,14 +129,92 @@
  *  - id optional, id that is assigned to the top level dialog node, but not the backdrop
  *  - variant: See PF4 Modal component's 'variant' property
  *  - titleIconVariant: See PF4 Modal component's 'titleIconVariant' property
+ *  - footerProps:
+ *      - cancel_clicked optional
+ *          Callback called when the dialog is canceled
  */
 class Dialog extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            action_in_progress: false,
+            action_in_progress_promise: null,
+            action_progress_message: '',
+            action_progress_cancel: null,
+            action_canceled: false,
+        };
+        this.update_progress = this.update_progress.bind(this);
+        this.cancel_click = this.cancel_click.bind(this);
+    }
+
+    update_progress(msg, cancel) {
+        this.setState({ action_progress_message: msg, action_progress_cancel: cancel });
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19200-20230821-124816-2fdc0a03-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)